### ISSUE TITLE
Memory leak when running the test campaign

### ISSUE THREAD

**Issue by mlaval** [angular/angular#11071]
When running the test campaign in any browser, we can see that the memory consumption keeps growing up, without any effective GC.
In browsers with low memory available, this causes crashes or disconnections (e.g. mobile browsers in SauceLabs or BrowserStack).

To reproduce this issue, add the spec file below to the project and run `./test.sh browser` as usual.
Then, watch Chrome getting slower and slower, until crashing.
```
import {Component} from "@angular/core";
import {TestBed} from "@angular/core/testing";

export function main() {
  fdescribe('should leak memory', () => {
    beforeEach(() => {
      TestBed.configureTestingModule({declarations: [TestComponent]});
    });

    for (var i = 0; i < 10000; i++) {
      it('should leak memory', () => {
        let fixture = TestBed.createComponent(TestComponent);
      });
    }
  });
}

@Component({selector: 'test-cmp', template: '<div></div>'})
class TestComponent {}
```

**Comment by kara** [comment]
@mlaval  Can this be closed, given #11072?

**Comment by mlaval** [comment]
Sure, fixed  in #11072
